tooling/ast_fuzzer/src/compare/interpreted.rs

```diff
@@ -192,6 +198,18 @@ impl Comparable for ssa::interpreter::errors::InterpreterError {
                 // So instead of reasoning about the `lhs` and `rhs` formats, let's just compare the message so we know it's the same constraint:
                 msg1 == msg2
             }
+            (
+                RangeCheckFailedWithMessage { message: msg1, .. },
+                ConstrainEqFailed { msg: msg2, .. },
+            ) => {
+                // The removal of unreachable instructions evaluates constant binary operations and can replace
+                // e.g. a `mul` followed by a `range_check` with a `constrain true == false, "attempt to multiple with overflow"`
+                msg2.contains(msg1)
+            }
+            (DivisionByZero { .. }, ConstrainEqFailed { msg, .. })
+            | (ConstrainEqFailed { msg, .. }, DivisionByZero { .. }) => {
+                msg == "attempt to divide by zero"
```

Comment:
Two things to note:
1. With the `RangeCheckFailedWithMessage` above I only considered going from a the `range_check` to `constrain`, because it's not going to be reversed. 
2. The `msg` of `ConstrainEqFailed` and `ConstrainNeFailed` I think would actually be `", \"attempt to divide by zero\""`, which is why I used `msg2.contains(msg1)`.

Reply:
This was an unfortunate hack I did in so `#[error("constrain {lhs_id} == {rhs_id}{msg} failed:\n    {lhs} != {rhs}")]` looks right, although I'm sure `this_error` has some trick to substitute a function call, so that the `msg` field can be nice and the display can be formatted.